controllers/compassmanager_controller.go

```diff
@@ -106,7 +110,23 @@ func (cm *CompassManagerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{RequeueAfter: cm.requeueTime}, err
 	}
 
+	kymaAnnotations, err := cm.getKymaAnnotations(req.NamespacedName)
+	if err != nil {
+		cm.Log.Warnf("Failed to obtain annotations from Kyma resource %s: %v.", req.Name, err)
+		return ctrl.Result{RequeueAfter: cm.requeueTime}, err
```

Comment:
`RequeueAfter` doesn't work if error was returned from `Reconcile`. Consider this flow
- if the error is from the k8s `client-go` method e.g. `Get`, `List`, `Update` the controller should set itself to Error state
- if the error results from any other component such as a problem with connection to Compass Director etc, the request should be requeueed in `RequeueAfter` with `nil` error